Modify project to remove KernelAbstractions, add CUDA to deps, remove StaticArrays

akshaysridhar opened the pull request:
Towards GPU compatibility

Removes KernelAbstractions and CUDAKernels from Project.toml
Removes StaticArrays from deps.
Adds CUDA to Project.
Adds compat entries for relevant packages.
Applies same changes to test environment.

charleskawczynski commented on src/SurfaceFluxes.jl:
```diff
@@ -416,31 +415,31 @@ function obukhov_length(
-                KA.@print("\n")
-                KA.@print("ts_sfc(sc) = ", ts_sfc(sc))
-                KA.@print("\n")
+                @printf("maxiter reached in SurfaceFluxes.jl:\n")
```
The KernelAbstractions `@print` is GPU compatible. And my guess is that`@printf` is not

akshaysridhar replied:
Right - for now this PR switches from KA -> CUDA + ClimaComms, the idea was to re-introduce the GPU print in the next PR. Would [CUDA.@cuprintf](https://docs.juliahub.com/CUDA/oWw5k/1.2.1/api/kernel/#CUDA.@cuprintf) this be a reasonable alternative ?

charleskawczynski replied:
Maybe, but the KernelAbstractions version adapts between CPU and GPU

akshaysridhar committed:
modified:   Project.toml
	modified:   src/SurfaceFluxes.jl
	modified:   test/test_profiles.jl
	modified:   test/test_convergence.jl

akshaysridhar commented:
Closing - the changes suggested in this PR have been broken up into smaller pieces.
